api/src/controllers/user/getUser.ts

```diff
@@ -0,0 +1,46 @@
+import { Request, Response } from 'express';
+
+import NotAuthorizedException from '../../exceptions/NotAuthorizedException';
+import * as userRepository from '../../repositories/userRepository';
+import { decodeQueryToUser } from '../../util/helper';
+import controller from '../controllerUtil';
+import Validator, { beProperlyUriEncoded } from '../validation';
+
+type ReqParams = {
+    user: string;
+};
+
+class ReqParamsValidator extends Validator<ReqParams> {
+    constructor() {
+        super('path parameters');
+
+        this.ruleFor('user').mustAsync(beProperlyUriEncoded);
+    }
+}
+
+type ResBody = {
+    exists: boolean;
+};
+
+/**
+ * Get my resume reviews.
+ * @param req HTTP request.
+ * @param res HTTP response.
+ * @returns Your resume reviews.
+ */
+const getUser = controller(async (req: Request<ReqParams, ResBody, unknown, unknown>, res: Response<ResBody>): Promise<void> => {
+    await new ReqParamsValidator().validateAndThrow(req.params);
+
+    const requested = decodeQueryToUser(req.params.user);
+    const requester = req.user.sub;
+
+    if (requester !== requested) {
+        throw new NotAuthorizedException();
+    }
+
+    const matches = await userRepository.get(requested);
+
+    res.status(200).json({ exists: matches.length == 1 });
```

Comment:
Like you suggested I would probably return the user object.

If the user is not registered I might return a status code 404 but a null user object also works.

I also like the idea of `/users/me`. The `{user}` doesn't add much.

Reply:
Agreed. I'm not sure how axios handles 404 but I do like that it better represents `NotFoundError` so let's go with 404 for now and update if necessary